### DISCUSSION TITLE
understanding-the-dotnet-ecosystem-the-introduction-of-dotnet-standard/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#129] [Announcements]
# understanding-the-dotnet-ecosystem-the-introduction-of-dotnet-standard/

In this article, previously part of my new book, we look at .NET Standard, look at why it was created, and discuss its future.

https://andrewlock.net/understanding-the-dotnet-ecosystem-the-introduction-of-dotnet-standard/

**Comment by chrisxfire** [comment]
That [metaphor](https://github.com/dotnet/standard/blob/release/3.0/docs/metaphor.md) from David Fowler you shared is gold and helped me to understand the relationship between .NET Standard and the various .NET Platforms.  Thank you!

**Comment by prlcutting** [comment]
Hi Andrew. Thanks for another great blog post.

You mentioned the following:

> .NET Standard only remains useful when you need to share code between .NET 5+ applications and legacy (.NET Core, .NET Framework, Xamarin) applications.

Perhaps a fringe use case, but source generators _require_ that you use .NET Standard, i.e. not .NET (as much as I wish it weren't the case). I know you know that, since you've blogged extensively on the subject! An awesome series of blog posts by the way.
